python/tskit/tables.py

```diff
@@ -3020,6 +3026,99 @@ def union(
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
 
+    def time_join(self, recent, time, continue_nodes=None):
+        ancient_ts = self.tree_sequence()
+        recent_ts = recent.tree_sequence()
+        if continue_nodes is None:
+            continue_nodes = ancient_ts.samples()
+
+        new_tables = recent_ts.tables
+        new_nodes = np.empty(0, dtype=np.uint32)
```

Comment:
It's actually probably more efficient to use a Python list here, since we're appending one-by-one, i.e.
``new_nodes = []``.